core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -101,6 +119,25 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                 if (AUTO.asString().equalsIgnoreCase(fuzziness)) {
                     return AUTO;
                 }
+                else if (fuzziness.toUpperCase(Locale.ROOT).startsWith(AUTO.asString() + ":")) {
+                    String[] fuzzinessLimit = fuzziness.substring(AUTO.asString().length() + 1).split(",");
+                    if (fuzzinessLimit.length == 2) {
+                        try {
+                            int lowerLimit = Integer.parseInt(fuzzinessLimit[0]);
+                            int highLimit = Integer.parseInt(fuzzinessLimit[1]);
+
+                            if (lowerLimit < 0 || highLimit < 0 || lowerLimit > highLimit) {
+                                throw new ElasticsearchParseException("Auto fuzziness wrongly configured");
+                            }
+                            return new Fuzziness("AUTO", lowerLimit, highLimit);
```

Comment:
it works for me. My point was more about keeping the low/high values in the string that we pass to `Fuzziness`, ie. `s/"AUTO"/fuzziness/`. It is important so that the low and high bounds are not lost upon serialization since this string is all that is written on the wire if this Fuzziness object needs to be sent to another node (see the `writeTo` method)